src/Paket.Core/Versioning/SemVer.fs

```diff
@@ -47,42 +53,72 @@ type PreRelease =
     { Origin : string
       Name : string
       Values : PreReleaseSegment list }
-    
+      
     static member TryParse (str : string) = 
-        if String.IsNullOrEmpty str then None 
+        if String.IsNullOrEmpty str then None
         else
-            let name = Regex("^(?<name>[a-zA-Z]+)").Match(str).Value
-
+            let getName fromList =
+                match fromList with
+                | AlphaNumeric(a)::_ -> a
+                | _::AlphaNumeric(a)::_ -> a // fallback to 2nd
```

Comment:
fallback to 2nd place, for cases where users want to move from non-compliant versions like `1.2.3.4-beta1` to valid semver2 like `1.2.3-4.beta.1` to keep similar precedence rules